problem-solver/cxx/check_student_answer_module/agent/check_student_answer_agent.cpp

```diff
@@ -0,0 +1,90 @@
+#include "check_student_answer_agent.hpp"
+
+#include "keynodes/check_student_answer_keynodes.hpp"
+
+namespace dm
+{
+
+ScAddr CheckStudentAnswerAgent::GetActionClass() const
+{
+  return CheckStudentAnswerKeynodes::action_check_student_answer;
+}
+
+ScResult CheckStudentAnswerAgent::DoProgram(ScAction & action)
+{
+  ScAddr const & studentAnswerAddr = action.GetArgument(ScKeynodes::rrel_1);
+  if (!m_context.IsElement(studentAnswerAddr)) {
+    m_logger.Warning("Action does not have one of arguments(or all).");
+    return action.FinishWithError();
+  }
+  ScTemplate templ1;
+  templ1.Quintuple(
+    action,
+    ScType::VarCommonArc,
+    ScType::VarNodeStructure >> "_structure",
+    ScType::VarPermPosArc,
+    CheckStudentAnswerKeynodes::nrel_expected_answer
+  );
+  ScTemplateSearchResult templ1Result;
+  m_context.SearchByTemplate(templ1, templ1Result);
+  if (templ1Result.Size() != 1) {
+    m_logger.Warning("Action does not have(or more than one) connection with expected answer.");
+    return action.FinishWithError();
+  }
+  ScAddr const & expectedAnswerAddr = templ1Result[0]["_structure"];
+  
+  ScAction someAction = m_context.GenerateAction(CheckStudentAnswerKeynodes::action_compare_structures);
+  ScAddr const & rrel1Addr = m_context.SearchElementBySystemIdentifier(ScKeynodes::rrel_1);
+  ScAddr const & rrel2Addr = m_context.SearchElementBySystemIdentifier(ScKeynodes::rrel_2);
+  ScAddr const & arc1 = m_context.GenerateConnector(ScType::ConstPermPosArc, someAction, studentAnswerAddr);
+  m_context.GenerateConnector(ScType::ConstPermPosArc, rrel1Addr, arc1);
+  ScAddr const & arc2 = m_context.GenerateConnector(ScType::ConstPermPosArc, someAction, expectedAnswerAddr);
+  m_context.GenerateConnector(ScType::ConstPermPosArc, rrel2Addr, arc2);
```

Comment:
```suggestion
  ScAddr const & arc1 = m_context.GenerateConnector(ScType::ConstPermPosArc, someAction, studentAnswerAddr);
  m_context.GenerateConnector(ScType::ConstPermPosArc, ScKeynodes::rrel_1, arc1);
  ScAddr const & arc2 = m_context.GenerateConnector(ScType::ConstPermPosArc, someAction, expectedAnswerAddr);
  m_context.GenerateConnector(ScType::ConstPermPosArc, ScKeynodes::rrel_2, arc2);
```
такое должно работать